packages/keyring/src/pair/index.ts

```diff
@@ -65,8 +65,7 @@ const verify = (type: KeypairType, message: Uint8Array, signature: Uint8Array, p
 export default function createPair (type: KeypairType, { publicKey, secretKey }: PairInfo, meta: KeyringPair$Meta = {}, encoded: Uint8Array | null = null): KeyringPair {
   return {
     type,
-    address: (): string =>
-      encodeAddress(publicKey),
+    address: encodeAddress(publicKey),
```

Comment:
There is an issue with this - if the prefix changes, this is not taken into account for the address. SWo basically, depending on where the set is called by the user, if may be incorrect.

Reply:
Think I am actually wrong here - passed by ref, so changes should reflect. Would probably be good adding an explicit test around this then. 

1.  Get address, compare to known 42 prefix 
2. Set prefix to 68
3. Compare to known 68 prefix

Reply:
So this is the test that (is not available, but would worin in master, but doesn't after this -

```js
// in pair/index.spec.ts

// once master is merged this import would be @polkadot/util-crypto
import { setAddressPrefix } from '../address';

describe('pair', () => {
  ...
  it('allows encoding of address with different prefixes', () => {
    expect(keyring.alice.address).toEqual(
      '5GoKvZWG5ZPYL1WUovuHW3zJBWBP5eT8CbqjdRY4Q6iMaQua'
    );

    setAddressPrefix(68);

    expect(keyring.alice.address).toEqual(
      '7sGUeMak588SPY2YMmmuKUuLz7u2WQpf74F9dCFtSLB2td9d'
    );
  });
```